### PR TITLE
ENHANCEMENT Ensure all base folders are excluded

### LINKED ISSUE #2059: Add reserved keywords
Such as `assets`, `vendor` and `resources`, these paths are now pretty much hardcoded to be blocked or redirected/handled elsewhere.
Top-level pages with these keywords should give a validation error or warning to keep users informed of what happened to their page.

# PRs:

 - [ ] https://github.com/silverstripe/silverstripe-cms/pull/2077

### PR DESCRIPTION
**Pull request by tractorcow**
Fixes #2059

### CODE REVIEW

File: code/Model/SiteTree.php [hunk 1]
```diff
@@ -1614,36 +1614,27 @@ public function validate()
...
-
-        if (!$this->ParentID && in_array($this->URLSegment, $disallowedSegments)) {
-            // Default to '-2', onBeforeWrite takes care of further possible clashes
-            return false;
-        }
-
-
-        if (self::config()->nested_urls && $parent = $this->Parent()) {
+            // Check if urlsegment matches a class name
+            if (class_exists($this->URLSegment) && is_subclass_of($this->URLSegment, RequestHandler::class)) {
```
Comment (flamerohr) [inline code/Model/SiteTree.php]: I thought Controller urlsegment access was removed during alpha?

Comment (tractorcow) [reply code/Model/SiteTree.php]: Yes but this seems a prudent check in any case.

I was worried if I removed it, it would be a risk.

**Comment by tractorcow** [comment]
I think I can remove it, now. I'll do it after lunch. ;0

**Comment by tractorcow** [comment]
Updated @flamerohr

**Pull request merged**